apps/authenticator/public/index.html

```diff
@@ -0,0 +1,93 @@
+    <pre id="status">authenticating...</pre>
+    <script>
+      function sendToOpener(value) {
+        window.opener.postMessage(
+          { from: 'popup', name: 'code', value: value },
+          '*'
+        )
+        window.close()
+      }
+
+      function fail() {
+        // pause before updating status, to make it seem like we're doing something
+        setTimeout(function() {
+          document.getElementById('status').innerText = 'cannot authenticate'
+        }, 500)
+      }
+
+      var code = new URLSearchParams(window.location.search).get('code')
+      console.log('code: ' + code)
+
+      // assuming window.opener is the Projects app, `pathname` will look like `/ipfs/Qmblahblahblah`
+      // we need to check that the `Qmblablahblah` part is a known version of Projects
+      var callingApp = window.opener && window.opener.location.pathname.split('/')[2]
```

Comment:
This line doesn't work with `localhost` due to same-origin restrictions.

Reply:
Also doesn't work when `opener` refers to an app in an iframe, because the iframe's origin is hidden from the popup. If the Aragon client included `allow-same-origin` on the iframe's `sandbox` property, we might be able to get around this, but that comes with significant security implications and is [not under consideration](https://github.com/aragon/aragon/blob/fcab70b3a30d6e5a4e33c2382f6cbc4c917da31a/src/security/configuration.js#L22).